### PR TITLE
Remove unused dependencies and upgrade to latest versions

### PR DESCRIPTION
**Pull request by ureeves**
We take a moment to check for unused dependencies in the crate, and upgrade when applicable.

### CODE REVIEW

**Review by herr-seppia — COMMENTED** [review]
Remove nested use

File: piecrust/src/store/tree.rs [hunk 1]
```diff
@@ -4,7 +4,10 @@
 //
 // Copyright (c) DUSK NETWORK. All rights reserved.
 
-use std::collections::{BTreeMap, BTreeSet};
+use std::{
+    cell::Ref,
+    collections::{BTreeMap, BTreeSet},
+};
```
Comment (herr-seppia) [inline piecrust/src/store/tree.rs:10]: Remove this nested use

File: piecrust/src/store.rs [hunk 2]
```diff
@@ -430,7 +431,12 @@ fn write_commit_inner<P: AsRef<Path>>(
             bytecode_dir,
             memory_dir,
             base: base.map(|inner| {
-                let base_root = inner.index.root();
+                let base_root_ref = inner.index.root();
+                let base_root = *base_root_ref;
+
+                // We manually drop the reference to the root hash to allow us
+                // to move out of `inner` when we return.
+                drop(base_root_ref);
```
Comment (herr-seppia) [inline piecrust/src/store.rs:439]: Isn't `let base_root = *inner.index.root();` enough to release the ref to base_root?

**Review by herr-seppia — APPROVED** [review]

**Review by moCello — APPROVED** [review]
LGTM

File: piecrust/src/store.rs (hunk 2 above)
Comment (ureeves) [reply piecrust/src/store.rs]: You're right, fixed.

File: piecrust/src/store/tree.rs (hunk 1 above)
Comment (ureeves) [reply piecrust/src/store/tree.rs:10]: We use it in other parts of the code here in piecrust, and I tend to prefer it, so I think I'll keep it.

**Pull request merged**